flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendMigrationTestBase.java

```diff
@@ -593,13 +598,14 @@ public void testStateBackendRestoreFailsIfNewKeySerializerRequiresMigration() th
             fail("should have failed");
         } catch (Exception expected) {
             // the new key serializer requires migration; this should fail the restore
-            Assert.assertTrue(
-                    ExceptionUtils.findThrowable(expected, StateMigrationException.class)
-                            .isPresent());
+            assertThat(
+                            ExceptionUtils.findThrowable(expected, StateMigrationException.class)
+                                    .isPresent())
+                    .isTrue();
         }
     }
 
-    @Test
+    @TestTemplate
     public void testStateBackendRestoreSucceedsIfNewKeySerializerRequiresReconfiguration()
```

Comment:
```suggestion
    void testStateBackendRestoreSucceedsIfNewKeySerializerRequiresReconfiguration()
```
could be package private